src/Microsoft.ML.Core/Data/ReadOnlyMemoryUtils.cs

```diff
@@ -0,0 +1,399 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Runtime.Internal.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.ML.Runtime.Data
+{
+    public static class ReadOnlyMemoryUtils
+    {
+
+        /// <summary>
+        /// This implements IEquatable's Equals method.
+        /// </summary>
+        public static bool Equals(ReadOnlyMemory<char> b, ReadOnlyMemory<char> memory)
+        {
+            if (memory.Length != b.Length)
+                return false;
+
+            Contracts.Assert(memory.IsEmpty == b.IsEmpty);
+
+            for (int i = 0; i < memory.Length; i++)
+            {
+                if (memory.Span[i] != b.Span[i])
+                    return false;
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Compare equality with the given system string value.
+        /// </summary>
+        public static bool EqualsStr(string s, ReadOnlyMemory<char> memory)
```

Comment:
>EqualsStr [](start = 27, length = 9)

Is this different method necessary due to perf reasons of `s.AsMemory`?

Reply:
You can say that.

---
In reply to: [216126482](https://github.com/dotnet/machinelearning/pull/863#discussion_r216126482) [](ancestors = 216126482)

Reply:
Would you say that?

---
In reply to: [216189037](https://github.com/dotnet/machinelearning/pull/863#discussion_r216189037) [](ancestors = 216189037,216126482)